### ISSUE TITLE
CF checks, check_time_coordinate, regex match result is None

### ISSUE THREAD

**Issue by FObersteiner** [ioos/compliance-checker#1214]
This
https://github.com/ioos/compliance-checker/blob/ff410d3ed9841fad4cd24fd6c6d11b127cc04847/compliance_checker/cf/cf_1_9.py#L112C1-L116C25

just gave me an error, caught as a warning:
```
WARNING: The following exceptions occurred during the cf:1.10 checker (possibly indicate compliance checker issues):
cf:1.10.check_time_coordinate: 'NoneType' object has no attribute 'group'
```
The input file specified the 'units' of the time coordinate as 'seconds since 1970-01-01' instead of expected by the regex 'seconds since 1970-01-01 00:00:00'.

Might be a version thing but in recent versions of Python, [re.match](https://docs.python.org/3/library/re.html#re.match) returns `None` if there is no match, not an AttributeError. So I think the code quoted should look more like
```py
            match = regex.match(seconds_regex, time_var.units)
            if match is None:
                continue
```

**Comment by ocefpaf** [comment]
It seems that it should be None since Python 2.7. The `AttributeError` is there for the `time_var.units` absence, not the `match`. My guess is that your data has `time_var.units`, but it failed to match somehow. We should catch that, but I wonder what do you have in `time_var.units` there.

**Comment by benjwadams** [comment]
@FObersteiner, could you provide a link to your data or a minimal example to reproduce this condition?

**Comment by FObersteiner** [comment]
First of all, please apologize, I should have provided exemplary data in the OP. Here's a file that reproduces this issue as well as #1215 : <https://thredds.atmohub.kit.edu/thredds/catalog/iagos-caribic/tmp/catalog.html?dataset=iagos-caribic/tmp/WSM_19971126_008_MLE_MUC_V03.nc> Note: this is work-in-progress, this link will very likely not work anymore in the future.

> The `AttributeError` is there for the `time_var.units` absence

OK, that wasn't obvious to me from the code. 

The data is created with the xarray Python library and definitely has the time_var.units attribute (my OP was incorrect there, it's not the 'calendar' that's causing the issue). The value of the units attribute is "seconds since 1970-01-01". xarray cuts the hh:mm:ss if they're all zero (still ISO8601-conform). The regex doesn't match this though, as it's looking for the 'ss'. I think it would be safe to assume hh:mm:ss == 00:00:00 if not provided, no? Just as CF assumes UTC in general.

**Comment by ocefpaf** [comment]
> I think it would be safe to assume hh:mm:ss == 00:00:00 if not provided, no? [CF conventions 1.12, 4.4.1](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.12/cf-conventions.html#time-coordinate-units) says

As I suspected it is more a "bug" in the regex. This particular check is checking if seconds is < 60 s. We should change to "when present it should be < 60 s." Does that make sense @benjwadams ?

**Comment by benjwadams** [comment]
Yes, I'll fix this.

**Comment by FObersteiner** [comment]
I think this is fixed via #1216; thanks a lot for the quick response!
